src/plugins/identity/README.md

```diff
@@ -10,10 +10,132 @@ GitHub and Discourse (potentially with a different username on each service).
+scores being generated. When including historical Graphs/Cred scores, for
+example to distribute Grain, you may need to contract the identities again.
+This can be tricky to do correctly, as they may have already received Grain
+on their GitHub/Discourse addresses before contraction.
+
+Code-reviewed support for Grain built into SourceCred should handle this in
+the future. But for now, requires the instance maintainer to handle Grain
+account contraction.
+
+## Configuration
+
+Identities can be configured using the `project.json` file.
+By adding entries to the the `identities` array.
+
+As an example of what this would look like:
+
+```js
+[
+  {
+    "type": "sourcecred/project",
+    "version": "0.4.0",
+  },
+  {
+    // The identities we should collapse.
+    "identities": [
+      {
+        "username": "User-A",
+        "aliases": ["github/user-a", "discourse/user-a"],
+      },
+      {
+        "username": "User-B",
+        "aliases": ["github/user-b", "discourse/user-b"],
+      },
+    ],
+
+    "id": "@example-project",
+    "discourseServer": {
+      // ...
+    },
+    "repoIds": [
+      // ...
+    ],
+  },
+]
+```
+
+### Username
+
+The `username` field allows you to choose a new username for the collapsed
+identity.
+
+It must be _unique in this project_ and can contain `A-Z`, `a-z`, `0-9`, `-` or
+`_` as valid characters.
+
+For display purposes the username is prefixed with an `@`.
+If a username is prefixed with `@` in the project file, it will be ignored.
+Meaning `"username": "User-A"` and `"username": "@User-A"` are equivalent.
+
+### Aliases
+
+Aliases have a format of:
+`{plugin}/{user ID for this plugin}`
+
+Currently supported plugins are:
+
+| Plugin | Example alias | Notes |
+|---|---|---|
+| GitHub | `github/example-user` | A user account on GitHub's platform. |
+| Discourse | `discourse/example-user` | A user account on the Discourse server configured in this project. |
+
+
+## How a contraction works
+
+Firstly, contractions are done _after_ the plugins have created their repective
+parts of the graph. But _before_ any Cred scores are calculated.
+
+The GitHub and Discourse plugin create edges for their respective user nodes
+before contraction:
+
+```
+GitHub user @abc
+├── authored Pull Request #123
+└── authored Issue #456
+
+Discourse user /u/abc
+├── authored Post /t/123/1
+└── authored Post /t/123/2
+```
+
+A new node is created, represeting the Identity, with the username
+provided in the project file. While removing the alias nodes.
+
+```
+Identity @abc
+
+???
+├── authored Pull Request #123
+└── authored Issue #456
+
+???
+├── authored Post /t/123/1
+└── authored Post /t/123/2
+```
+
+And we connect the edges to the new identity node.
+
+```
+Identity @abc
+├── authored Pull Request #123
+├── authored Issue #456
+├── authored Post /t/123/1
+└── authored Post /t/123/2
+```
+
+Now we're ready to calculate Cred scores.
+
+### Reference detection
```

Comment:
This is the first time we are talking about "the URL" - maybe we can have a sentence like "A sourcecred prototype creates a URL to represent each identity. For example, for identity `abc` we would have:
```
<show url>
```
However the urls are maintained for the previous alias for reference. For example, we can refer to any of these URLS

```
<show urls>
```
and they will take us to the page for `<previous url for abc>`

Reply:
> A sourcecred prototype creates a URL to represent each identity.

It does not actually do that though :]
There's currently no URL to refer to the identity.

Reply:
okay, whatever it does, it would be helpful to explain the space of these URLs because this is the first time it's being mentioned. My lack of understanding is a reflection of that missing bit.

Reply:
What do you mean by "the space of these URLs"? Do you mean where the URLs occur?

Since this is about reference detection, any place the other plugins apply reference detection. Like Discourse posts, GH comments or commit messages.

Reply:
This section is still not clear to me, I am not familiar with the name space of URLs.

Reply:
The section "Reference Detection" opens and I need a little more description for what reference detection is, and why those URLs are relevant to it. This is written from the point of someone that understands these concepts, it doesn't take account for the reader that does not.

Reply:
AFAIK the identity plugin doesn't interact with reference detection at all. Basically the rule is that the identity plugin will re-write edges to identities, which will cover reference edges that came from individual plugins too. Given this, is there a need to mention reference detection explicitly here at all?

Reply:
@decentralion that might be the best solution! If we keep some mention here, the description that you just wrote is very clear and could be used.

Reply:
Yes, technically it only does the contraction. So it doesn't interact with reference detection directly.

Reason I'm mentioning it is because I'm anticipating a question. For example when creating an initiative json file where URLs are currently the only accepted input.

Q: Now `@abc` has an identity, and identities don't have URLs, how should I refer to that identity as a URL?
A: Use the URL of any of the underlying aliases, as the contraction happens after reference detection, it will refer to the identity.

I could add this example as a caveat for initiatives' docs instead. But sided with caution to ensure there's something on the subject rather than nothing :]

Reply:
Maybe that should go with documentation for creating the json file (and then be linked from here?)

Reply:
Agree with @vsoch. Was confused as to why reference detection was discussed here. 

It's good to have this topic documented somewhere, but I think there's a better place for it. It feels like more developer-centric doco that should live in more low-level technical documentation, or even just in code comments for now. Perhaps the Initiatives plugin README if that helps explain why you can use aliases to point to contributors.

Reply:
Got it, I'll remove this section and leave it for the Initiatives plugin to explain.